### PR TITLE
feat(layout): introduce @zendeskgarden/css-layout package

### PR DESCRIPTION
**Pull request by austin94**
## Description

This PR introduces the `@zendeskgarden/css-layout` package. This package is an opinionated wrapper around the [Bootstrap v4 Flexbox Grid](https://getbootstrap.com/docs/4.1/layout/grid/).

The customizations are:

* Gutter size of 20px
* `is-rtl` class to apply RTL styling (🎉 yay flexbox)
* `debug` class to apply blue styling on columns to more easily debug grid sizes 

## Detail

Pre-published at: https://garden.zendesk.com/css-components/grid/

screen shot 2018-05-14 at 12 38 00 pm

## Checklist

* [x] :ok_hand: style updates are Garden Designer approved (add the
  designer as a reviewer)
* [x] :globe_with_meridians: component demo is up-to-date (`yarn start`)
* [x] :white_check_mark: all component states are represented
  (`.is-hovered`, `.is-focused`, etc.)
* [x] :arrow_left: renders as expected with reversed (RTL) direction
* [x] :metal: renders as expected sans Bedrock (`?bedrock=false`)
* [x] :nail_care: provides `custom.css` example for modifying the
  primary accent color
* [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: demo/grid/index.html [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+          </div>
+        </div>
+      </div>
+      <h2 class="u-gamma u-mb">RTL Locales</h2>
+      <p class="u-mb">
+        You can enable RTL layouts by using the <code class="c-code">.is-rtl</code> class with the
+        <code class="c-code">.container</code> or <code class="c-code">.container-fluid</code>
+        wrapper classes</a>.
+      </p>
+      <div class="container-fluid is-rtl debug u-mb-xl">
```
Comment (jzempel) [inline demo/grid/index.html]: This section doesn't make sense after a user has toggled the page's `RTL` on/off. Maybe just move the paragraph text up to the top and call out the ability to toggle RTL on the page.

File: demo/grid/index.html [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+            3 of 3
+          </div>
+        </div>
+      </div>
+      <section id="playground">
+        <h1 class="c-main__subtitle u-alpha u-mb">Playground</h1>
+        <div class="c-playground">
+          <div class="c-playground__preview"></div>
+<textarea class="c-playground__code">
+<div class="container-fluid debug">
```
Comment (jzempel) [inline demo/grid/index.html]: make this section "spaceless" with respect to the `<textarea>`. No leading/trailing whitespace. See other demo pages for details.

File: packages/grid/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# @zendeskgarden/css-grid
+
+This package contains an opinionated version of the Bootstrap v4 flexbox grid.
+
+It includes:
+
+* Zendesk specific gutter sizing
+* RTL locale support with the `.is-rtl` class
+* Debug styling support with the `.debug` class
```
Comment (jzempel) [inline packages/grid/README.md]: Rename `.is-debug` (per below).

File: packages/grid/src/index.scss [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ * RTL locale styling
+ */
+.container.is-rtl, .container-fluid.is-rtl {
+  direction: rtl;
+}
+
+/**
+ * Debug styling for columns
+ */
+.debug {
```
Comment (jzempel) [inline packages/grid/src/index.scss]: Let's rename this to `.is-debug` to assert consistency with the other boolean variables.

File: packages/grid/src/index.scss [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  .row > .col, .row > [class^='col-'] {
+    border: 1px solid $zd-color-blue-400;
```
Comment (jzempel) [inline packages/grid/src/index.scss]: Consider applying an alpha 35% in order to smooth out adjoining borders a bit.

File: demo/grid/index.html [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+        <svg class="c-header__logo__svg svg-square">
+          <use xlink:href="../index.svg#zd-svg-icon-26-zendesk" />
+        </svg>
+      </a
+    ></figure
+    ><h1 class="c-header__title"
+      ><a class="u-fg-inherit" href="..">CSS Components</a>
+      <span>/ Grid</span>
+    </h1>
+    <ul class="c-ctl c-ctl--header">
```
[the remark this answers is not in the document]
Comment (austin94) [reply demo/grid/index.html]: Fixed.

File: demo/grid/index.html [hunk 7]
```diff
@@ -0,0 +1,165 @@
+<!doctype html>
+<html>
+<head>
+  <title>Grid / CSS Components / Zendesk Garden</title>
+  <link href="../bedrock/index.css" rel="stylesheet">
+  <link href="../menus/index.css" rel="stylesheet">
```
[the remark this answers is not in the document]
Comment (austin94) [reply demo/grid/index.html]: Removed.

File: demo/grid/index.html (hunk 1 above)
Comment (austin94) [reply demo/grid/index.html]: Moved.

File: demo/grid/index.html (hunk 2 above)
Comment (austin94) [reply demo/grid/index.html]: Fixed.

File: packages/grid/README.md (hunk 3 above)
Comment (austin94) [reply packages/grid/README.md]: Renamed.

File: packages/grid/src/index.scss (hunk 4 above)
Comment (austin94) [reply packages/grid/src/index.scss]: Renamed.

File: packages/grid/src/index.scss (hunk 5 above)
Comment (austin94) [reply packages/grid/src/index.scss]: Added to both background and border.

File: packages/grid/package.json [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    "build": "node-sass --include-path ../../node_modules --include-path node_modules --output dist src/index.scss",
+    "format": "prettier-package-json --write",
+    "lint": "stylelint $(find src -name '*.scss')",
+    "prepack": "cp ../../LICENSE.md .",
+    "test": "yarn lint",
+    "watch": "yarn build --watch"
+  },
+  "devDependencies": {
+    "@zendeskgarden/css-variables": "^4.1.3",
+    "bootstrap": "^4.1.1",
```
Comment (ryanseddon) [inline packages/grid/package.json]: These should be in dependencies

File: packages/grid/src/index.scss [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+ */
+.container.is-rtl, .container-fluid.is-rtl {
+  direction: rtl;
+}
+
+/**
+ * Debug styling for columns
+ */
+.is-debug {
+  .row > .col, .row > [class^='col-'] {
```
Comment (ryanseddon) [inline packages/grid/src/index.scss]: You can just drop the hyphen from your attribute selector and do this:

```css
.is-debug {
  .row > [class^='col'] {
  //...
  }
}
```

**Pull request merged**